.github/workflows/ci.yml

```diff
@@ -49,114 +51,132 @@ jobs:
 
       - name: Run type check
         run: npm run typecheck
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' && !startsWith(github.ref, 'refs/heads/release/') }}
```

Comment:
👀  This ensures that ci jobs from PRs are cancelled when new commits are pushed, except for `main` and release branches. This should reduce queuing times.